.pre-commit-config.yaml

```diff
@@ -0,0 +1,34 @@
+minimum_pre_commit_version: "2.9.0"
```

Comment:
We should add the pre-commit install process as an install step in the `contributing.md`

Reply:
We can, though fwiw I don't plan to install it personally.  I plan to add it to CI

Reply:
I added a blurb to contributor docs linking to `pre-commit` docs in case people want to learn how to install hooks / run them manually locally